doit/dependency.py

```diff
@@ -361,6 +338,96 @@ def remove_all(self):
         self._conn.execute('delete from doit')
 
 
+class BaseChecker(object):
+    """Base Checker that must be inherited."""
+
+    def __init__(self, backend):
+        self.backend = backend
+        self._set = self.backend.set
+        self._get = self.backend.get
+
+    def check_modified(self, file_path, state):
+        """Check if file in file_path is modified from previous "state".
+
+        @param file_path (string): file path
+        @param state (tuple), timestamp, size, md5
```

Comment:
The state (timestamp, size, md5) is an implementation detail of the current implementation...

We happens if a new checker needs to save different information?
What if the user runs `doit` once with one checker, and then again with a different checker?